app/scripts/controllers/network/network.js

```diff
@@ -136,20 +134,24 @@ export default class NetworkController extends EventEmitter {
     // Ping the RPC endpoint so we can confirm that it works
     const ethQuery = new EthQuery(this._provider)
     const initialNetwork = this.getNetworkState()
-    ethQuery.sendAsync({ method: 'net_version' }, (err, _networkVersion) => {
+    ethQuery.sendAsync({ method: 'net_version' }, (err, networkVersion) => {
       const currentNetwork = this.getNetworkState()
       if (initialNetwork === currentNetwork) {
         if (err) {
           this.setNetworkState('loading')
           return
         }
 
-        // Now we set the network state to the chainId computed earlier
-        this.setNetworkState(chainId)
+        this.setNetworkState(networkVersion)
```

Comment:
This still constitutes a change in behavior, as beforehand we used the network version for the built-in Infura networks, but we used the `chainId` field for custom RPC networks. This is relied upon in various places in our codebase, such as for address book entries.

Reply:
This was fixed in 8b5b254